lms/services/lti_role_service.py

```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+from sqlalchemy.orm import Session
+
+from lms.models import LTIRole
+
+
+class LTIRoleService:
+    """A service for dealing with LTIRole objects."""
+
+    def __init__(self, db_session: Session):
+        self._db = db_session
+
+    def get_roles(self, role_description: str) -> List[LTIRole]:
+        """
+        Get a list of role objects for the provided strings.
+
+        :param role_description: A comma delimited set of role strings
+        """
+        role_strings = [role.strip() for role in role_description.split(",")]
+
+        # pylint: disable=no-member
+        # Pylint is confused about the `in_` for some reason
+        roles = self._db.query(LTIRole).filter(LTIRole.value.in_(role_strings)).all()
+
+        if missing := (set(role_strings) - set(role.value for role in roles)):
```

Comment:
I wonder if an upsert would be faster/shorter

This is not a problem in any case.

Reply:
Yes it could be. I was thinking that this will end up being basically read only, and this gets it to only one query (in that case), but an upsert would probably be the same regardless...

I suspect this would be faster than an upsert, but I didn't actually bench it.